Roma Class1 - Week 1 till Week 3 (week-3-practise not included)

Flaminiat opened the pull request.

jslopezgithub commented on week-1-practise/level-2/4-currency-conversion-2.js:
```diff
@@ -6,9 +6,16 @@
   They have also decided that they should add a 1% fee to all foreign transactions
   Find a way to add 1% to all currency conversions (think about the DRY principle)
 */
+function allForeign(price) {
```
Good idea. You could have extended this idea to pass the rate as well and simply your two methods further.
 
```javascript
function allForeign(price, rate) {
    return parseFloat(price * rate + price /100)).toFixed(2);
}
```

jslopezgithub commented on week-1-practise/level-3/1-piping.js:
```diff
@@ -11,7 +11,11 @@
 
 var startingValue = 5;
+function allInOne(num) {
+  return parseFloat(num * 2 + 10).toFixed(2);
```
The exercise was asking to do this using the provided functions: `add`, `multiply` and `format`.

jslopezgithub commented on week-2-practise/level-1/2-validation.js:
```diff
@@ -2,7 +2,13 @@
 // TIP: You should write an expression that returns a boolean value
 
 var num = 10;
-var isBigEnough; // ONLY EDIT THIS LINE
```
In this exercise you should have only edited this line. i.e. `var isBigEnough = number < 8;`

jslopezgithub commented on week-2-practise/level-2/2-validation.js:
```diff
@@ -4,7 +4,13 @@
 
-function validate(num) {}
+function validate(num) {
+  if (typeof num == "number" && num > 0 && num <= 100) {
```
This is correct, but it can be simplified.
```suggestion
  return typeof num == "number" && num > 0 && num <= 100;
```

jslopezgithub commented on week-2-practise/level-3/6-map.js:
```diff
@@ -11,11 +11,15 @@ function tidyUpString(str) {
 }
 
 function captialise(str) {
+  var restOfTheName = tidyUpString(str).split("");
```
Another solution
```suggestion
function captialise(str) {
  return str[0].toUpperCase() + str.slice(1);
  // complete this function
}
```

jslopezgithub commented on week-2/F-logical-operators/exercise.js:
```diff
@@ -11,26 +11,43 @@ var cssLevel = 4;
+var htmlLevelAbove5 = verification(htmlLevel);
+var cssLevelAbove5 = verification(cssLevel);
+function verification(num) {
+  if (num > 5) {
```
```suggestion
return num > 5;
```

jslopezgithub commented on week-2/G-conditionals-2/exercise-1.js:
```diff
@@ -7,9 +7,12 @@
 
 function negativeOrPositive(number) {
-
+  if (number < 0) {
```
This is a good place user the ternary operator
```
return number < 0 ? "negative" : "positive";
```

jslopezgithub commented on week-2/G-conditionals-2/exercise-1.js:
```diff
@@ -7,9 +7,12 @@
-
+  if (number < 0) {
+    return "negative";
+  } else if (number >= 0) {
```
You could have use here an else statement.

jslopezgithub commented on week-2/G-conditionals-2/exercise-2.js:
```diff
@@ -8,7 +8,11 @@
-
+  if (grade >= 50) {
+    return "passed"
+} else if(grade < 50)  {
```
No need for else if here, else would have worked.

jslopezgithub commented on week-2/I-array-properties/exercise.js:
```diff
@@ -6,9 +6,12 @@
 */
 
 function isEmpty(arr) {
```
```suggestion
function isEmpty(arr) {
  return arr.length == 0; // complete this statement
}
```

jslopezgithub commented on week-2/J-array-get-set/exercise.js:
```diff
@@ -5,11 +5,11 @@
 
 function last(arr) {
-  return; // complete this statement
+  return arr[2]; // complete this statement
```
This should be based on the array size.
```suggestion
function last(arr) {
  return arr[arr.length - 1]; // complete this statement
}
```

Flaminiat commented:
thank you, well received! :)
Have a good day,
Flaminia

Il giorno gio 11 lug 2019 alle ore 17:00 jslopezgithub <
notifications@github.com> ha scritto:

> *@jslopezgithub* approved this pull request.
>
> Good job overall. Make it a point to avoid doing this.
> if (true) { return true; } else { return false; }
>
> When checking for a condition, you can simply return it. i.e. return true.
> i.e. return i > 5
> ------------------------------
>
> In week-1-practise/level-2/4-currency-conversion-2.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302581607>
> :
>
> > @@ -6,9 +6,16 @@
>
>    They have also decided that they should add a 1% fee to all foreign transactions
>
>    Find a way to add 1% to all currency conversions (think about the DRY principle)
>
>  */
>
> +function allForeign(price) {
>
>
> Good idea. You could have extended this idea to pass the rate as well and
> simply your two methods further.
>
> function allForeign(price, rate) {
>
>     return parseFloat(price * rate + price /100)).toFixed(2);
>
> }
>
> ------------------------------
>
> In week-1-practise/level-3/1-piping.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302582784>
> :
>
> > @@ -11,7 +11,11 @@
>
>  */
>
>
>
>  var startingValue = 5;
>
> +function allInOne(num) {
>
> +  return parseFloat(num * 2 + 10).toFixed(2);
>
>
> The exercise was asking to do this using the provided functions: add,
> multiply and format.
> ------------------------------
>
> In week-2-practise/level-1/2-validation.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302585038>
> :
>
> > @@ -2,7 +2,13 @@
>
>  // TIP: You should write an expression that returns a boolean value
>
>
>
>  var num = 10;
>
> -var isBigEnough; // ONLY EDIT THIS LINE
>
>
> In this exercise you should have only edited this line. i.e. var
> isBigEnough = number < 8;
> ------------------------------
>
> In week-2-practise/level-2/2-validation.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302585603>
> :
>
> > @@ -4,7 +4,13 @@
>
>  // - is less than or equal to 100
>
>  // Tip: write other small functions for each requirement
>
>
>
> -function validate(num) {}
>
> +function validate(num) {
>
> +  if (typeof num == "number" && num > 0 && num <= 100) {
>
>
> This is correct, but it can be simplified.
> ⬇️ Suggested change
>
> -  if (typeof num == "number" && num > 0 && num <= 100) {
>
> +  return typeof num == "number" && num > 0 && num <= 100;
>
>
> ------------------------------
>
> In week-2-practise/level-3/6-map.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302586559>
> :
>
> > @@ -11,11 +11,15 @@ function tidyUpString(str) {
>
>  }
>
>
>
>  function captialise(str) {
>
> +  var restOfTheName = tidyUpString(str).split("");
>
>
> Another solution
> ⬇️ Suggested change
>
> -  var restOfTheName = tidyUpString(str).split("");
>
> +function captialise(str) {
>
> +  return str[0].toUpperCase() + str.slice(1);
>
> +  // complete this function
>
> +}
>
>
> ------------------------------
>
> In week-2/F-logical-operators/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302586972>
> :
>
> > @@ -11,26 +11,43 @@ var cssLevel = 4;
>
>
>
>  // Finish the statement to check whether HTML, CSS knowledge are above 5
>
>  // (hint: use the comparison operator from before)
>
> -var htmlLevelAbove5;
>
> -var cssLevelAbove5;
>
> +var htmlLevelAbove5 = verification(htmlLevel);
>
> +var cssLevelAbove5 = verification(cssLevel);
>
> +function verification(num) {
>
> +  if (num > 5) {
>
>
> ⬇️ Suggested change
>
> -  if (num > 5) {
>
> +return num > 5;
>
>
> ------------------------------
>
> In week-2/G-conditionals-2/exercise-1.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302587697>
> :
>
> > @@ -7,9 +7,12 @@
>
>  */
>
>
>
>  function negativeOrPositive(number) {
>
> -
>
> +  if (number < 0) {
>
>
> This is a good place user the ternary operator
>
> return number < 0 ? "negative" : "positive";
>
>
> ------------------------------
>
> In week-2/G-conditionals-2/exercise-1.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302587809>
> :
>
> > @@ -7,9 +7,12 @@
>
>  */
>
>
>
>  function negativeOrPositive(number) {
>
> -
>
> +  if (number < 0) {
>
> +    return "negative";
>
> +  } else if (number >= 0) {
>
>
> You could have use here an else statement.
> ------------------------------
>
> In week-2/G-conditionals-2/exercise-2.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302588066>
> :
>
> > @@ -8,7 +8,11 @@
>
>  */
>
>
>
>  function studentPassed(grade) {
>
> -
>
> +  if (grade >= 50) {
>
> +    return "passed"
>
> +} else if(grade < 50)  {
>
>
> No need for else if here, else would have worked.
> ------------------------------
>
> In week-2/I-array-properties/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302588451>
> :
>
> > @@ -6,9 +6,12 @@
>
>  */
>
>
>
>  function isEmpty(arr) {
>
>
> ⬇️ Suggested change
>
> - function isEmpty(arr) {
>
> +function isEmpty(arr) {
>
> +  return arr.length == 0; // complete this statement
>
> +}
>
>
> ------------------------------
>
> In week-2/J-array-get-set/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/380#discussion_r302588717>
> :
>
> >  }
>
>
>
>  function last(arr) {
>
> -  return; // complete this statement
>
> +  return arr[2]; // complete this statement
>
>
> This should be based on the array size.
> ⬇️ Suggested change
>
> -  return arr[2]; // complete this statement
>
> +function last(arr) {
>
> +  return arr[arr.length - 1]; // complete this statement
>
> +}
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/js-exercises/pull/380?email_source=notifications&email_token=ALJIR47PEI7YFQK2P7DNO6DP65DHBA5CNFSM4IBDSQIKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB6FLDUI#pullrequestreview-260747729>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALJIR4ZWOXYAEG4ECMHZOXTP65DHBANCNFSM4IBDSQIA>
> .
>
